### DISCUSSION TITLE
New update to 2.12 has broken where.OR.contact usage - How to update?

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#4380] [Q&A] [answered]
Hi,

Something I was doing prior to updating was building a query like so:

```ts
async getThreads (userId: number, args?: MessageThreadGetArgs): Promise<MessageThreadDto[]> {
    let where: Prisma.MessageThreadWhereInput = {}
    where.OR = []

    if (args !== void 0) {
      // Get threads which are new / read / blocked.
      if (args.state !== void 0) {
        where.OR = where.OR.concat([
          { states: { some: { userId, state: args.state } } },
          { messages: { some: { state: args.state } } }
        ])
      }
})
```

Now, TS won't compile because of:

> Property 'concat' does not exist on type '(Without<MessageThreadWhereInput, Enumerable<MessageThreadWhereInput>> & MessageThreadWhereInput[]) | (Without<...> & MessageThreadWhereInput)'.

I've tried various casts etc but I'm not quite sure what I need to do here to dynamically build this array of OR clauses with the new typings. Prisma's typing has always gone over my head and it's just got a little more crazy :)

Would appreciate some help on this.

**Comment by webnoob** [comment, marked as answer]
Hmm, this seems to satisfy it (although, yet untested) ...

```ts
where.OR = [
  ...(where.OR as []),
  { states: { some: { userId, state: args.state } } },
  { messages: { some: { state: args.state } } }
]
```

Would appreciate some feedback on the correct course though.

EDIT: Tested and working correctly.

**Reply by ryands17** [reply]
@webnoob 👋 
This is due to the changes in the recent types due to the introduction of [setting foreign keys directly](https://github.com/prisma/prisma/releases/tag/2.11.0).

For e.g. if I have a schema like this:

```prisma
model User {
  id    Int    @id @default(autoincrement())
  name  String
  email String
}
```

The types for `where.OR` are generated like this (you will be able to check those in the generated types):

![image](https://user-images.githubusercontent.com/19697099/100436505-76843b00-30c5-11eb-8e9e-43dbf9b73f7e.png)

This XOR type was added which has changed the way you declare your types, which is why you would need to use the above workaround.

**Reply by webnoob** [reply]
Yes, thanks. That makes sense. I did see the changes in there and thought it was because of that so thanks for the clarification and I'm pleased the work around is valid.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
